lib/archethic/mining/pending_transaction_validation.ex

```diff
@@ -275,6 +173,54 @@ defmodule Archethic.Mining.PendingTransactionValidation do
     end
   end
 
+  defp validate_contract(%Transaction{data: %TransactionData{code: code}})
+       when byte_size(code) > @code_max_size do
+    {:error, "invalid contract type transaction , code exceed max size"}
+  end
+
+  @spec validate_ownerships(Transaction.t()) :: :ok | {:error, any()}
+  defp validate_ownerships(%Transaction{data: %TransactionData{ownerships: []}}), do: :ok
+
+  defp validate_ownerships(%Transaction{data: %TransactionData{ownerships: ownerships}}) do
+    Enum.reduce_while(ownerships, :ok, fn
+      ownership, :ok ->
+        %Ownership{secret: secret, authorized_keys: authorized_keys} = ownership
+        nb_authorized_keys = map_size(authorized_keys)
+
+        cond do
+          secret == "" ->
+            {:halt, {:error, "invalid transaction - Ownership: secret is empty"}}
+
+          nb_authorized_keys == 0 ->
```

Comment:
Here you could just pattern match the map `authorized_keys == %{}` it will be more efficient than calculting the size of the map